### PR TITLE
Add Support For Container Image Multi-Arch Builds

### LINKED ISSUE #113: ARM Container Images
I would like to run the scheduler-plugins on ARM(specifically ARM64) hardware. I'd like an official scheduler-plugins container image for this use case. This would be in addition to the already provided x86_64 container image.

### PR DESCRIPTION
**Pull request by seanmalloy**
Initially support amd64 and arm64 container images. Additional
architectures can be added in the future if needed.

Fixes #113

### CODE REVIEW

**Comment by seanmalloy** [comment]
/kind feature
/hold

I am able to successfully build and push the AMD64 and ARM64 images to a registry. I have not actually tested the images in a cluster.

**Comment by seanmalloy** [comment]
I finally tested that the new images work. Images confirmed working on a kind cluster on my laptop.

/hold cancel
/cc denkensk yuanchen8911

**Comment by denkensk** [comment]
Thanks so much.
/lgtm

File: Makefile [hunk 1]
```diff
@@ -40,29 +41,66 @@ all: build
...
+	docker build -f ./build/scheduler/Dockerfile --build-arg ARCH="amd64" --build-arg RELEASE_VERSION="$(RELEASE_VERSION)" -t $(RELEASE_REGISTRY)/$(RELEASE_IMAGE)-amd64 .
+	docker build -f ./build/controller/Dockerfile --build-arg ARCH="amd64" -t $(RELEASE_REGISTRY)/$(RELEASE_CONTROLLER_IMAGE)-amd64 .
 
-.PHONY: release-image
-release-image: clean
-	docker build -f ./build/scheduler/Dockerfile --build-arg RELEASE_VERSION="$(RELEASE_VERSION)" -t $(RELEASE_REGISTRY)/$(RELEASE_IMAGE) .
-	docker build -f ./build/controller/Dockerfile -t $(RELEASE_REGISTRY)/$(RELEASE_CONTROLLER_IMAGE) .
+.PHONY: release-image.arm64v8
+release-image.arm64v8: clean
+	docker build -f ./build/scheduler/Dockerfile --build-arg ARCH="arm64v8" --build-arg RELEASE_VERSION="$(RELEASE_VERSION)" -t $(RELEASE_REGISTRY)/$(RELEASE_IMAGE)-arm64 .
```
Comment (Huang-Wei) [inline Makefile:86]: Does this mean the eventual docker images postfixed with `arm64` is based on a `arm64v8/alpine:3.12` image?

File: Makefile [hunk 2]
```diff
@@ -40,29 +41,66 @@ all: build
...
 .PHONY: build-controller
 build-controller: autogen
 	$(COMMONENVVAR) $(BUILDENVVAR) go build -ldflags '-w' -o bin/controller cmd/controller/controller.go
 
+.PHONY: build-controller.amd64
+build-controller.amd64: autogen
+	$(COMMONENVVAR) $(BUILDENVVAR) GOARCH=amd64 go build -ldflags '-w' -o bin/controller cmd/controller/controller.go
+
+.PHONY: build-controller.arm64v8
+build-controller.arm64v8: autogen
```
Comment (Huang-Wei) [inline Makefile:59]: Can this be combined with `build-controller.amd64`?

File: Makefile [hunk 3]
```diff
@@ -40,29 +41,66 @@ all: build
...
 .PHONY: build-scheduler
 build-scheduler: autogen
 	$(COMMONENVVAR) $(BUILDENVVAR) go build -ldflags '-X k8s.io/component-base/version.gitVersion=$(VERSION) -w' -o bin/kube-scheduler cmd/scheduler/main.go
 
+.PHONY: build-scheduler.amd64
+build-scheduler.amd64: autogen
+	$(COMMONENVVAR) $(BUILDENVVAR) GOARCH=amd64 go build -ldflags '-X k8s.io/component-base/version.gitVersion=$(VERSION) -w' -o bin/kube-scheduler cmd/scheduler/main.go
+
+.PHONY: build-scheduler.arm64v8
+build-scheduler.arm64v8: autogen
```
Comment (Huang-Wei) [inline Makefile:71]: Ditto. It seems the command is identical with `build-scheduler.amd64`.

**Review by Huang-Wei — COMMENTED** [review]
Thanks @seanmalloy ! Some nits.

File: Makefile (hunk 1 above)
Comment (seanmalloy) [reply Makefile:86]: Yes, that is correct.

File: Makefile (hunk 2 above)
Comment (seanmalloy) [reply Makefile]: Fixed in commit 9ca52707ffc1826e7ce53f45c710e00ce0b279ce.

File: Makefile (hunk 3 above)
Comment (seanmalloy) [reply Makefile]: Fixed in commit 9ca52707ffc1826e7ce53f45c710e00ce0b279ce.

**Comment by seanmalloy** [comment]
@Huang-Wei and @denkensk I have addressed the review feedback. Please let me know if any additional changes are needed. Thanks!

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks @seanmalloy !

**Pull request merged**